subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuitesMultiTargetIntegrationTest.groovy

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.testsuites
+
+import org.gradle.api.internal.tasks.testing.junit.JUnitTestFramework
+import org.gradle.api.plugins.jvm.internal.DefaultJvmTestSuite
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class TestSuitesMultiTargetIntegrationTest extends AbstractIntegrationSpec {
+    def "multiple targets can be used"() {
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            ${mavenCentralRepository()}
+
+            testing {
+                suites {
+                    test {
+                        useJUnit()
+                        targets {
+                            test17 {
+                                testTask.configure {
+                                    javaLauncher = javaToolchains.launcherFor {
+                                        languageVersion = JavaLanguageVersion.of(17)
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            [tasks.test, tasks.test17].each { task ->
+                task.doLast {
+                    assert testFramework instanceof ${JUnitTestFramework.canonicalName}
+                    assert classpath.size() == 2
+                    assert classpath.any { it.name == "junit-${DefaultJvmTestSuite.TestingFramework.JUNIT4.getDefaultVersion()}.jar" }
+                }
+            }
+        """
+
+        when:
+        var result = succeeds("check")
```

Comment:
You don't need the `var result = ` here because `succeeds` already sets `result` from the integration test base class